src/persist/src/indexed/unsealed.rs

```diff
@@ -123,22 +123,30 @@ impl Unsealed {
     ) -> Result<UnsealedBatchMeta, Error> {
         let key = Unsealed::new_blob_key();
         let desc = batch.desc.clone();
-        let ts_upper = match batch.updates.last() {
-            Some(upper) => upper.1,
-            None => {
-                return Err(Error::from(
-                    "invalid unsealed batch: trying to write empty batch",
-                ))
-            }
-        };
-        let ts_lower = match batch.updates.first() {
-            Some(lower) => lower.1,
-            None => {
-                return Err(Error::from(
-                    "invalid unsealed batch: trying to write empty batch",
-                ))
+        if batch.updates.is_empty() {
+            return Err(Error::from(
+                "invalid unsealed batch: trying to write empty batch",
+            ));
+        }
+
+        let (ts_upper, ts_lower) = {
+            let mut upper = Timestamp::minimum();
+            let mut lower = u64::MAX;
```

Comment:
nit: this get a little more bulletproof if you store upper and lower as an Option of a 2-tuple as you iterate. and then move the `return Err(... "invalid unsealed batch: trying to write empty batch")` to cover the case where you're done iterating and the Option is still None